src/explorer/appServiceNode.ts

```diff
@@ -89,7 +89,50 @@ export class AppServiceNode extends NodeBase {
         return this.getTreeDataProvider<AppServiceDataProvider>().azureAccount;
     }
 
+    async localGitDeploy(azureAccount): Promise<boolean> {
+        const publishCredentials = await this.getWebSiteManagementClient(azureAccount).webApps.listPublishingCredentials(this.site.resourceGroup, this.site.name);
+        const config = await this.getWebSiteManagementClient(azureAccount).webApps.getConfiguration(this.site.resourceGroup, this.site.name);
+        const oldDeployment = await this.getWebSiteManagementClient(azureAccount).webApps.listDeployments(this.site.resourceGroup, this.site.name);
+        if (config.scmType !== 'LocalGit') {
+            let input = await window.showErrorMessage(`Local Git Deployment is not set up. Set it up in the Azure Portal.`, `Go to Portal`)
+            if (input === 'Go to Portal') {
+                const deepLink = `https://ms.portal.azure.com/#resource${config.id.substring(0, config.id.indexOf('/config/web'))}/DeploymentSource`
+                opn(deepLink);
+            }
+            return;
+        }
+
+        const username = publishCredentials.publishingUserName;
+        const password = publishCredentials.publishingPassword;
+        const repo = `${this.site.enabledHostNames[1]}:443/${this.site.repositorySiteName}.git`;
+        const remote = `https://${username}:${password}@${repo}`;
+
+        let git = require('simple-git/promise')(workspace.rootPath);
+
+        await git.init()
+            .catch(err => {
+                window.showErrorMessage(`Git must be installed to use Local Git Deploy.`, `Install`)
+                    .then(input => {
+                        if (input === 'Install') {
+                            opn(`https://git-scm.com/downloads`);
+                        }
+                    });
+                return;
+            })
+        git.add('./*');
+        git.commit('Deployed through VS Code');
```

Comment:
Per offline discussion, maybe we don't need this commit? We should just push whatever the user has already committed

However, I'm not sure how that will work if the user didn't have a git repository initialized (since they'll have nothing to push unless we commit). We might have to discuss that behavior with Chris or Matt